watertap/unit_models/electrodialysis_1D.py

```diff
@@ -1556,6 +1691,162 @@ def eq_dl_thickness(self, mem, side, t, x):
+            units=pyunits.meter**2 * pyunits.second**-1,
+            doc="The mass diffusivity of the solute as molecules (not individual ions)",
+        )
+        self.hydraulic_diameter = Var(initialize=1e-3, units=pyunits.meter)
+        self.N_Re = Var(
+            initialize=100, units=pyunits.dimensionless, doc="Reynolds Number"
+        )
+        self.N_Sc = Var(
+            initialize=2000, units=pyunits.dimensionless, doc="Schmidt Number"
+        )
+        self.N_Sh = Var(
+            initialize=100, units=pyunits.dimensionless, doc="Sherwood Number"
+        )
+
+        if self.config.hydraulic_diameter_method == HydraulicDiameterMethod.fixed:
+            _log.warning("Do not forget to FIX the channel hydraulic diameter in [m]!")
+        else:
+
+            @self.Constraint(
+                doc="To calculate hydraulic diameter",
+            )
+            def eq_hydraulic_diameter(self):
+                if (
+                    self.config.hydraulic_diameter_method
+                    == HydraulicDiameterMethod.conventional
+                ):
+                    return (
+                        self.hydraulic_diameter
+                        == 2
+                        * self.channel_height
+                        * self.cell_width
+                        * self.spacer_porosity
+                        * (self.channel_height + self.cell_width) ** -1
+                    )
+                else:
+                    self.spacer_specific_area = Var(
+                        initialize=1e4, units=pyunits.meter**-1
+                    )
+                    return (
+                        self.hydraulic_diameter
+                        == 4
+                        * self.spacer_porosity
+                        * (
+                            2 * self.channel_height**-1
+                            + (1 - self.spacer_porosity) * self.spacer_specific_area
+                        )
+                        ** -1
+                    )
+
+        @self.Constraint(
+            doc="To calculate Re",
+        )
+        def eq_Re(self):
+
+            return (
+                self.N_Re
+                == self.dens_mass
+                * self.velocity_D[0, 0]
+                * self.hydraulic_diameter
+                * self.visc_d**-1
+            )
+
+        @self.Constraint(
+            doc="To calculate Sc",
+        )
+        def eq_Sc(self):
+
+            return (
+                self.N_Sc == self.visc_d * self.dens_mass**-1 * self.diffus_mass**-1
+            )
+
+        @self.Constraint(
+            doc="To calculate Sc",
+        )
+        def eq_Sh(self):
+
+            return self.N_Sh == 0.29 * self.N_Re**0.5 * self.N_Sc**0.33
+
+    def _pressure_drop_calculation(self):
+        self.pressure_drop = Var(
+            self.flowsheet().time,
+            initialize=1e4,
+            units=pyunits.pascal * pyunits.meter**-1,
+            doc="pressure drop per unit of length",
```

Comment:
Currently you have a single pressure drop (pressure/length) for the whole module, our RO model calculates this locally for each node (I don't think this PR should necessarily change the calculation to each node, but it could be considered for future PRs, or we should justify why ED shouldn't). 

Additionally, this pressure drop is only calculated based on diluate velocity, but you created a variable for the concentrate velocity. Is this a mistake? For our OARO model we calculate the pressure drop separately for each channel (feed and permeate side), it may make sense for ED as well.

Reply:
Thanks @TimBartholomew .  Yes, the current model assumes a constant velocity when calculating N_Re and friction pressure, although the actual velocity has ~4% change when water migration is molded. The error caused by this change for friction pressure will be even smaller because v is squared in its calculation.  I accept this approximation by the judging this error level is much insignificant compared to errors in calculating others, e.g., hydraulic diameter, and friction factor (which can an error of >10%).  I can note this in the later documentation update . 

Another thing to consider is that,  if velocity change is fully considered in accounting for pressure change, I think the correct way is to apply a Navier-Stokes or Bernoullie's equation to expressing energy and momentum conservations of the flow.  The velocity change in my case corresponds to a pressure change.   I haven't seen the two conservation equations be implemented in IDAES control-volumes and I think the approximation as-is is good for the purpose of the current model. 

Upon accepting the approximation, there is no difference between the flow in diluate and concentrate channels in the current ED setup because of identical channel dimensions.   So all the friction pressure in the model is taking ` velocity_diluate[ 0, 0]` (which is equivalent to `velocity_concentrate[0,0]`)  and passed to both channels.  The same strategy is used for control volume `length_domain`.  This is really just a mathematical workaround.  `velocity_concentrate` is indeed just calculated and unused for other things, but I think it's ok to let them hanging there as a calculation output.  Inspecting velocities might be useful in the future somewhere. 

In summary, my recommendation is to accept the constant velocity approximation for calculating pressure drop thus far.  Let me know if you have more thoughts.   Thanks for your comment that made me think it through more.

Reply:
@lbibl, this approximation has to be documented. Can you please open another ticket for this issue. If RO is including the pressure drop for each channel, ED should be be able to model this as well.

Reply:
Thanks for reviewing, @srikanthallu .  I will make sure updating the decimation to reflect key contents.  To clarify, pressure drops for both channels are accounted in the current.